bokehjs/src/coffee/util/zoom.coffee

```diff
@@ -0,0 +1,80 @@
+scale_range = ({frame, factor, center, h_axis_only = false, v_axis_only = false}) ->
```

Comment:
I *think* (always dangerous) that you shouldn't have to worry about stubbing any solver/canvas things because the computed ranges that you're dealing with are all model properties (and not view properties, which involve the solver and canvas stuff).

I added some tests to update the ranges for a wheel pan tool that you should be able to crib: https://github.com/bokeh/bokeh/blob/master/bokehjs/test/models/tools/gestures/wheel_pan_tool.coffee